controllers/providers/kubernetes/nodes.go

```diff
@@ -32,3 +38,45 @@ func (k *KubernetesClientSet) ListClusterNodes() (*corev1.NodeList, error) {
 	}
 	return nodes, nil
 }
+
+// DrainNode cordons and drains a node.
+func (k *KubernetesClientSet) DrainNode(node *corev1.Node, PostDrainDelaySeconds time.Duration, client kubernetes.Interface) error {
+	if client == nil {
+		return fmt.Errorf("K8sClient not set")
+	}
+
+	if node == nil {
+		return fmt.Errorf("node not set")
+	}
+
+	helper := &drain.Helper{
+		Client:              client,
+		Force:               true,
+		GracePeriodSeconds:  -1,
+		IgnoreAllDaemonSets: true,
+		Out:                 os.Stdout,
+		ErrOut:              os.Stdout,
+		DeleteEmptyDirData:  true,
+		Timeout:             time.Duration(120) * time.Second,
```

Comment:
What is the timeout we set in v1 controller? can we keep the same?

Reply:
Yes. I have to pull that from Spec. I just did